File: DailyChallenges/CPP/LeetCode_198.cpp
// 198. House Robber @ 2022/12/14

class Solution {
public:
    int rob(vector<int>& nums) {
        int t[2] = {0};
        for (int n: nums){
            int tmp = max(t[0] + n, t[1]);
            t[0] = t[1];
            t[1] = tmp;
        }
        return max(t[0], t[1]);
    }
};

File: DailyChallenges/Scala/LeetCode_198.scala
object Solution {
    def rob(nums: Array[Int]): Int = {
        val (s1,s2) = nums.foldLeft((0,0))((p, n) => (p._2, math.max(p._1 + n, p._2)))
        math.max(s1,s2)
    }
}
